server/app/auth/api_key.py

```
@@ -0,0 +1,32 @@
+from fastapi import Security, HTTPException, Depends
+from fastapi.security.api_key import APIKeyHeader
+from starlette.status import HTTP_403_FORBIDDEN
+from typing import Optional
+from app.core.config import get_settings, Settings
+
+API_KEY_NAME = "X-API-Key"
+api_key_header = APIKeyHeader(name=API_KEY_NAME, auto_error=False)
+
+async def get_api_key(
+    api_key_header: Optional[str] = Security(api_key_header),
+    settings: Settings = Depends(get_settings)
```

Ruff (0.8.0)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)